webapp/channels/src/components/copy_button.tsx

```diff
@@ -1,28 +1,53 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import type {Placement} from '@floating-ui/react';
 import classNames from 'classnames';
 import React, {useRef, useState} from 'react';
-import {Tooltip} from 'react-bootstrap';
-import {FormattedMessage} from 'react-intl';
+import {useIntl} from 'react-intl';
 
-import OverlayTrigger from 'components/overlay_trigger';
+import useTooltip from 'components/common/hooks/useTooltip';
 
-import Constants from 'utils/constants';
-import {t} from 'utils/i18n';
 import {copyToClipboard} from 'utils/utils';
 
 type Props = {
     content: string;
     beforeCopyText?: string;
     afterCopyText?: string;
-    placement?: string;
+    placement?: Placement;
     className?: string;
 };
 
 const CopyButton: React.FC<Props> = (props: Props) => {
     const [isCopied, setIsCopied] = useState(false);
     const timerRef = useRef<NodeJS.Timeout | null>(null);
+    const intl = useIntl();
+
+    const getDefaultMessage = () => {
+        if (isCopied) {
+            return props.afterCopyText;
+        }
+        return props.beforeCopyText ?? 'Copy code';
+    };
+
+    const getId = () => {
+        if (isCopied) {
+            return 'copied.message';
+        }
+        return props.beforeCopyText ? 'copy.text.message' : 'copy.code.message';
+    };
```

Comment:
we have to combine this into single function which returns the correct message back

Reply:
This will be resolved by updating this to the latest master. I replaced these functions with one that returns a `MessageDescriptor` object with `id` and `defaultMessage`